packages/block-library/src/navigation-menu/index.php

```diff
@@ -65,62 +63,75 @@ function build_css_colors( $attributes ) {
+	return '<nav class="wp-block-navigation-menu" style="' . esc_attr( $colors['style'] ) . '"><ul class="menu">' . walk_nav_menu_tree( $items, 0, $args ) . '</ul></nav>';
+}
 
-	$colors = build_css_colors( $attributes );
+/**
+ * Adds inline styles to menu item link attributes.
+ *
+ * @param array  $attributes Link attributes.
+ * @param object $item       Menu item.
+ * @return array
+ */
+function block_navigation_link_attributes( $attributes, $item ) {
+	if ( ! empty( $item->style ) ) {
+		$attributes['style'] = $item->style;
+	}
 
-	return "<nav class='wp-block-navigation-menu' {$comp_inline_styles}>" .
-		build_navigation_menu_html( $block, $colors ) .
-	'</nav>';
+	return $attributes;
 }
 
 /**
- * Walks the inner block structure and returns an HTML list for it.
- *
- * @param array $block  The block.
- * @param array $colors Contains inline styles and CSS classes to apply to menu item.
+ * Prepares menu items to be used in Walker_Nav_Menu.
  *
- * @return string Returns  an HTML list from innerBlocks.
+ * @param array $block  The parsed block.
+ * @param array $colors Custom colors classes and styles.
+ * @return array Menu items
  */
-function build_navigation_menu_html( $block, $colors ) {
-	$html = '';
-	foreach ( (array) $block['innerBlocks'] as $key => $menu_item ) {
-
-		$html .= '<li class="wp-block-navigation-menu-item ' . $colors['bg_css_classes'] . '"' . $colors['bg_inline_styles'] . '>' .
-			'<a
-				class="wp-block-navigation-menu-item__link ' . $colors['text_css_classes'] . '"
-				' . $colors['text_inline_styles'];
+function setup_block_nav_items( $block, $colors ) {
+	static $menu_item_id = 1;
+	$nav_menu_items = array();
+	$nav_menu_item  = array(
+		'classes'       => $colors['classes'],
+		'current'       => false,
+		'menu_order'    => 0,
+		'post_content'  => '',
+		'post_excerpt'  => '',
+		'post_type'     => 'nav_menu_item',
+		'style'         => $colors['style'],
+	);
 
-		if ( isset( $menu_item['attrs']['url'] ) ) {
-			$html .= ' href="' . $menu_item['attrs']['url'] . '"';
-		}
-		if ( isset( $menu_item['attrs']['title'] ) ) {
-			$html .= ' title="' . $menu_item['attrs']['title'] . '"';
+	foreach ( $block['innerBlocks'] as $inner_block ) {
+		$sub_menu_items = setup_block_nav_items( $inner_block, $colors );
+		foreach ( $sub_menu_items as $sub_menu_item ) {
+			$sub_menu_item->menu_item_parent = $menu_item_id;
+			$nav_menu_items[]                = $sub_menu_item;
 		}
-		$html .= '>';
-		if ( isset( $menu_item['attrs']['label'] ) ) {
-			$html .= $menu_item['attrs']['label'];
-		}
-		$html .= '</a>';
 
-		if ( count( (array) $menu_item['innerBlocks'] ) > 0 ) {
-			$html .= build_navigation_menu_html( $menu_item, $colors );
+		if ( empty( $inner_block['attrs']['label'] ) ) {
+			continue;
 		}
 
-		$html .= '</li>';
+		$nav_menu_item['ID']         = $menu_item_id;
+		$nav_menu_item['post_title'] = $inner_block['attrs']['label'];
+		$nav_menu_item['url']        = $inner_block['attrs']['destination'] ?? '#';
```

Comment:
Going to add a blocker because of this change.
The link of the menu item is stored [into the attributes object in the `url` property](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/navigation-menu/edit.js#L55).
If we want to change this name to `destination` we should either make the change in this same PR or create a new one.
Basically it doesn't apply the `href` property.

Reply:
Thanks! No I'm quite happy that it was changed to `url`, I didn't realize that was merged already.